### PR TITLE
SAT-126: Schedule an Item functionality added

### LINKED ISSUE #87: Schedule an Item
from the menu, and using the icon, this week's expanded view

### PR DESCRIPTION
**Pull request by thestoneroller**
### Changes Made

- **ScheduleItem.tsx**:
  - Show date suggestions

- Added **ScheduleInboxPopover.tsx**:

- **inbox.store.ts**:
  - Updated API response handling by changing `response.data.response.items` to `response.data.items` to fix an issue where the previous structure was undefined.

- **InboxActions**:
  - Removed unused file.

- **getUserItems (item.service.js)**:
  - Removed `dueDate: null` filter from the query to include items with any due date, not just null values.

### Checklist

- [x] Tested locally
- [x] Synced with the latest main branch

### Linked Issues:

fixes #87

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `ScheduleInboxPopover` component for improved scheduling functionality in the inbox.
	- Added a `ScheduleItem` component to manage and display scheduling information with date selection capabilities.

- **Bug Fixes**
	- Enhanced the handling of API responses in the `moveItemToDate` method for better data consistency.

- **Style**
	- Improved formatting and consistency in the codebase for better readability.

### CODE REVIEW

File: apps/backend/src/services/lib/item.service.js [hunk 1]
```diff
@@ -1,231 +1,236 @@
...
+    user: me,
+    isCompleted: false,
+    isArchived: false,
+    isDeleted: false,
+    spaces: { $exists: true, $eq: [] },
+    status: { $nin: ['archive', 'done'] },
+  }).sort({ createdAt: -1 });
+
+  return items;
+};
```
Comment (sajdakabir) [inline apps/backend/src/services/lib/item.service.js:16]: No changes are needed in getUserItems.

 In the inbox, we are displaying item that don't have any due dates.

**Review by sajdakabir — CHANGES_REQUESTED** [review]

**Comment by sajdakabir** [comment]
item.service had lint error

Screenshot 2024-10-17 at 1 39 39 PM

File: apps/backend/src/services/lib/item.service.js (hunk 1 above)
Comment (thestoneroller) [reply apps/backend/src/services/lib/item.service.js:16]: Oh, ok. I'll change

**Comment by thestoneroller** [comment]
> item.service had lint error
> Screenshot 2024-10-17 at 1 39 39 PM

where can i change the lint settings on my end?

**Comment by thestoneroller** [comment]
Why do I feel like the bot will keep suggesting changes forever and I'm gonna refactor the code forever 😭

**Pull request closed without merging**